src/main/io/serial_cli.h

```diff
@@ -20,6 +20,195 @@
 
 extern uint8_t cliMode;
 
+#ifndef USE_QUAD_MIXER_ONLY
+//  this with mixerMode_e
+static const char * const mixerNames[] = {
+    "TRI", "QUADP", "QUADX", "BI",
+    "GIMBAL", "Y6", "HEX6",
+    "FLYING_WING", "Y4", "HEX6X", "OCTOX8", "OCTOFLATP", "OCTOFLATX",
+    "AIRPLANE", "HELI_120_CCPM", "HELI_90_DEG", "VTAIL4",
+    "HEX6H", "PPM_TO_SERVO", "DUALCOPTER", "SINGLECOPTER",
+    "ATAIL4", "CUSTOM", "CUSTOMAIRPLANE", "CUSTOMTRI", NULL
+};
+#endif
+
+// sync this with features_e
+static const char * const featureNames[] = {
+    "RX_PPM", "VBAT", "INFLIGHT_ACC_CAL", "RX_SERIAL", "MOTOR_STOP",
+    "SERVO_TILT", "SOFTSERIAL", "GPS", "FAILSAFE",
+    "SONAR", "TELEMETRY", "CURRENT_METER", "3D", "RX_PARALLEL_PWM",
+    "RX_MSP", "RSSI_ADC", "LED_STRIP", "DISPLAY", "ONESHOT125",
+    "BLACKBOX", "CHANNEL_FORWARDING", NULL
+};
+
+static const char * const lookupTableOffOn[] = {
+    "OFF", "ON"
+};
+
+static const char * const lookupTableUnit[] = {
+    "IMPERIAL", "METRIC"
+};
+
+static const char * const lookupTableAlignment[] = {
+    "DEFAULT",
+    "CW0",
+    "CW90",
+    "CW180",
+    "CW270",
+    "CW0FLIP",
+    "CW90FLIP",
+    "CW180FLIP",
+    "CW270FLIP"
+};
+
+#ifdef GPS
+static const char * const lookupTableGPSProvider[] = {
+    "NMEA", "UBLOX"
+};
+
+static const char * const lookupTableGPSSBASMode[] = {
+    "AUTO", "EGNOS", "WAAS", "MSAS", "GAGAN"
+};
+#endif
+
+static const char * const lookupTableCurrentSensor[] = {
+    "NONE", "ADC", "VIRTUAL"
+};
+
+static const char * const lookupTableGimbalMode[] = {
+    "NORMAL", "MIXTILT"
+};
+
+static const char * const lookupTablePidController[] = {
+    "MW23", "MWREWRITE", "LUX"
+};
+
+static const char * const lookupTableBlackboxDevice[] = {
+    "SERIAL", "SPIFLASH"
+};
+
+static const char * const lookupTableSerialRX[] = {
+    "SPEK1024",
+    "SPEK2048",
+    "SBUS",
+    "SUMD",
+    "SUMH",
+    "XB-B",
+    "XB-B-RJ01",
+    "IBUS"
+};
+
+static const char * const lookupTableGyroFilter[] = {
+    "OFF", "LOW", "MEDIUM", "HIGH"
+};
+
+static const char * const lookupTableGyroLpf[] = {
+    "256HZ",
+    "188HZ",
+    "98HZ",
+    "42HZ",
+    "20HZ",
+    "10HZ"
+};
+
+
+typedef struct lookupTableEntry_s {
```

Comment:
you're right but this should be treated in a specific PR

Le 05/01/2016 12:33, Petr Ledvina a écrit :
>
> In src/main/io/serial_cli.h 
> <https://github.com/cleanflight/cleanflight/pull/1654#discussion_r48836104>:
>
> > +
> > +static const char * const lookupTableGyroFilter[] = {
> > +    "OFF", "LOW", "MEDIUM", "HIGH"
> > +};
> > +
> > +static const char * const lookupTableGyroLpf[] = {
> > +    "256HZ",
> > +    "188HZ",
> > +    "98HZ",
> > +    "42HZ",
> > +    "20HZ",
> > +    "10HZ"
> > +};
> > +
> > +
> > +typedef struct lookupTableEntry_s {
>
> When you are at it - it is possible to NULL-terminate all lookup 
> tables and calculate length at runtime (performance is not critical). 
> Then this indirection (and |enum|) is not necessary, pointer to first 
> item can be stored in cliValues.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/cleanflight/cleanflight/pull/1654/files#r48836104>.
>